### ISSUE TITLE
Error in "describe Unknown Feature Type" test

### ISSUE THREAD

**Issue by clausnagel** [opengeospatial/ets-wfs20#144]
The "describe Unknown Feature Type" test uses the following query to provide an unknown feature type for the `DescribeFeatureType` operation. It expects the WFS to respond with a HTTP status code 400.

```
?service=WFS&version=2.0.0&request=DescribeFeatureType&typenames=ns90:Unknown1.Type&namespaces=xmlns(xml,http://www.w3.org/XML/1998/namespace),xmlns(ns90,http://example.org),xmlns(wfs,http://www.opengis.net/wfs/2.0)
```
However, the query uses the `typenames` parameter which is not defined by the WFS specification. The parameter is called `typename` (without the trailing 's'). 

So, in my opinion, the WFS should ignore the unknown `typenames` parameter. And since the correct `typename` parameter is omitted, the complete application schema shall be returned (cf. chapter 9.2.4.1). And consequently, a HTTP status code of 200 must be send.

A correct WFS implementation will therefore fail this test.

**Comment by clausnagel** [comment]
Well, there is also another issue with this test IMHO.

The POST version runs fine because it uses the correct `<wfs:TypeName>` element. However, the test method `describeUnknownFeatureType` in the class `DescribeFeatureTypeTests` then checks for an `InvalidParameterValue` exception code. As references, cl. 8.3.4, 9.2.4.1 are listed on the method.

However, from these clauses I cannot read that an unknown feature type on a `DescribeFeatureType` operation shall raise an `InvalidParameterValue`. There is a similar requirement stated in ATS A.2.22.1.1 but this only refers to adhoc query expressions, not to `DescribeFeatureType`.

**Comment by lgoltz** [comment]
typenames (with trailing s) is correct for WFS 2.0 KVP/GET-requests. In WFS 1.1 and 1.0 the parameter was named typename.

**Comment by lgoltz** [comment]
@clausnagel Thanks for reporting this. 

typenames Parameter should be renamed to typename for DescribeFeatureType requests (I assume this also applies to other tests).

Specification/Test must be rechecked if  InvalidParameterValue is the expected exception.

**Comment by lgoltz** [comment]
Fix is solved by PR #150.

For the question in https://github.com/opengeospatial/ets-wfs20/issues/144#issuecomment-475183232 a new issue was created: #149

**Comment by dstenger** [comment]
Fix was successfully tested with service provided by @clausnagel.
Test `describe Unknown Feature Type` is passing.
